finagle-core/src/main/scala/com/twitter/finagle/dispatch/PipeliningDispatcher.scala

```diff
@@ -54,10 +55,14 @@ private[finagle] class PipeliningDispatcher[Req, Rep](
 
   loop()
 
+  protected def respond(q: Q, out: Try[Out]): Unit
+
+  protected def pipeline(req: Req, p: Promise[Rep]): Future[Q]
+
   // Dispatch serialization is guaranteed by GenSerialClientDispatcher so we
   // leverage that property to sequence `q` offers.
   protected def dispatch(req: Req, p: Promise[Rep]): Future[Unit] =
```

Comment:
I hesitated to do that and didn't similar to how `apply` is not final in `GenSerialClientDispatcher`. I figured that most use-cases would only need to override `pipeline`, but potentially more complex ones would need to override `dispatch` as well. I don't have a strong opinion, either way is fine by me.